plugins/pi4j-plugin-linuxfs/src/main/java/com/pi4j/plugin/linuxfs/provider/i2c/LinuxFsI2C.java

```diff
@@ -130,10 +137,27 @@ public int writeRegister(int register, byte[] data, int offset, int length) {
         return write(tmp);
     }
 
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public int writeRegister(byte[] register, byte[] data, int offset, int length) {
+        Objects.checkFromIndexSize(offset, length, data.length);
+        byte[] tmp = new byte[length + register.length];
+        for(int i = 0; i < register.length; i++){
```

Comment:
here you could use System.arrayCopy as well, no?

Reply:
will look at that.